Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -423,8 +444,394 @@ theorem sum_congr {f : SkewMonoidAlgebra k G} {M : Type*} [AddCommMonoid M] {g
     (h : ∀ x ∈ f.support, g₁ x (f.coeff x) = g₂ x (f.coeff x)) :
     f.sum g₁ = f.sum g₂ := Finset.sum_congr rfl h
 
+theorem induction_on {p : SkewMonoidAlgebra k G → Prop} (f : SkewMonoidAlgebra k G)
+    (zero : p 0) (single : ∀ g a, p (single g a)) (add : ∀ f g :
+    SkewMonoidAlgebra k G, p f → p g → p (f + g)) : p f := by
+  rw [← sum_single f, sum_def']
+  exact Finset.sum_induction _ _ add zero (by aesop)
+
+/--
+Slightly less general but more convenient version of `SkewMonoidAlgebra.induction_on`. -/
+theorem induction_on' [instNonempty : Nonempty G] {p : SkewMonoidAlgebra k G → Prop}
```

Comment:
```suggestion
@[induction_eliminator]
theorem induction_on' [instNonempty : Nonempty G] {p : SkewMonoidAlgebra k G → Prop}
```
It seems you want this as default eliminator for induction, but I am not sure. If you add it you can remove the various `using induction_on'` in the uses of the `induction` tactic.